CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+cmake_minimum_required(VERSION 3.21)
+
+project(
+  KART
+  VERSION 0.10.6
+  LANGUAGES C CXX)
+
+enable_testing()
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+set(KART_CMAKE_TEMPLATE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/cmake/template)
+
+include(PythonGetABIInfo)
+
+# find_package(Git REQUIRED) execute_process( COMMAND ${GIT_EXECUTABLE} rev-parse HEAD
+# RESULT_VARIABLE result OUTPUT_VARIABLE KART_GIT_COMMIT OUTPUT_STRIP_TRAILING_WHITESPACE )
+# if(result) message(FATAL_ERROR "Failed to get Git commit: ${result}") endif()
+
+#
+# options
+#
+
+option(IS_HERMETIC "Isolated build where Kart builds all dependencies")
+message("Hermetic build? ${IS_HERMETIC}")
+
+set(VENDOR_ARCHIVE
+    ""
+    CACHE
+      FILEPATH
+      "Use vendor packages from CI: path to vendor archive file from https://github.com/koordinates/kart"
+)
+
+#
+# setup
+#
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  set(LINUX ON)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  set(MACOS ON)
+endif()
+
+# CCache
+find_program(CCACHE_PROGRAM ccache)
+if(CCACHE_PROGRAM)
+  set(ccacheEnv CCACHE_CPP2=true CCACHE_BASEDIR=${CMAKE_BINARY_DIR}
+                CCACHE_SLOPPINESS=pch_defines,time_macros)
+  foreach(lang IN ITEMS C CXX)
+    set(CMAKE_${lang}_COMPILER_LAUNCHER ${CMAKE_COMMAND} -E env ${ccacheEnv} ${CCACHE_PROGRAM})
+  endforeach()
+endif()
+
+# OSX: prefer Homebrew over Frameworks
+set(Python3_FIND_IMPLEMENTATIONS "CPython")
+find_package(Python3 3.7 REQUIRED COMPONENTS Interpreter)
+pythongetabiinfo()
+
+# version configure_file(${KART_CMAKE_TEMPLATE_DIR}/VERSION.in kart/VERSION @ONLY)
```

Comment:
accidentally commented out?

Reply:
ah, I'll revisit it at some point. Will remove for now.